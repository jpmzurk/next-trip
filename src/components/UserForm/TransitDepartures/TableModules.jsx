import React from "react";
import {
  TableRow,
  TableHead,
  TableFooter,
  TableBody,
  IconButton,
  CircularProgress,
  TableCell,
} from "@mui/material/";
import "../FadeIn.css";
import "./Departures.css";
import AddCircleOutlineOutlinedIcon from "@mui/icons-material/AddCircleOutlineOutlined";
import RemoveCircleOutlineOutlinedIcon from "@mui/icons-material/RemoveCircleOutlineOutlined";

const headerStyles = {
  fontWeight: 700,
  color: "#626462",
  lineHeight: 1.1,
};
const subHeaderStyles = {
  fontWeight: 700,
  color: "#626462",
  fontSize: "1rem",
  letterSpacing: ".5px",
};

const bodyStyles = {
  color: "#626462",
  fontWeight: 500,
  fontSize: "1.2rem",
};

const DeparturesHeader = ({ __stopId }) => {
  return (
    <TableHead className="fadeIn">
      <TableRow>
        <TableCell
          sx={{ fontSize: "1.8rem", width: "80%", pl: 3, ...headerStyles }}
          colSpan={2}
        >
          {__stopId.StopLabel}
        </TableCell>
        <TableCell
          sx={{ fontSize: "1rem", pr: 3, ...headerStyles }}
          align="right"
        >
          <strong>Stop #</strong>: {__stopId.StopID}
        </TableCell>
      </TableRow>
      <TableRow className="tableSubheader">
        <TableCell sx={{ pl: 3, ...subHeaderStyles }}> Route </TableCell>
        <TableCell sx={{ ...subHeaderStyles }}>Destination</TableCell>
        <TableCell align="right" sx={{ pr: 3, ...subHeaderStyles }}>
          Departs
        </TableCell>
      </TableRow>
    </TableHead>
  );
};

function DeparturesBody({ departures }) {
  return (
    <TableBody className="fadeIn">
      {departures.map((departure) => (
        <TableRow key={departure.DepartureTime}>
          <TableCell name="route" sx={{ pl: 3, ...bodyStyles }}>
            {departure.Route}
          </TableCell>
          <TableCell name="stop" sx={{ width: 500, ...bodyStyles }}>
            {departure.Description}
          </TableCell>
          <TableCell
            name="departures"
            sx={{ pr: 3, ...bodyStyles }}
            align="right"
          >
            {departure.DepartureText}
          </TableCell>
        </TableRow>
      ))}
    </TableBody>
  );
}

const DeparturesFooter = ({ handleExpand, isExpanded }) => {
  return (
    <TableFooter className="fadeIn">
      <TableRow>
        <TableCell>
          <IconButton
            name="iconButton"
            aria-label="open or close"
            onClick={handleExpand}
          >
            {isExpanded ? (
              <>
                <RemoveCircleOutlineOutlinedIcon
                  aria-label="close"
                  name="collapse"
                  color="primary"
                />
              </>
            ) : (
              <>
                <AddCircleOutlineOutlinedIcon
                  aria-label="open"
                  name="expand"
                  color="primary"
                />
              </>
            )}
          </IconButton>
        </TableCell>
      </TableRow>
    </TableFooter>
  );
};

const NoDepartures = () => {
  return (
    <TableBody className="fadeIn">
      <TableRow>
        <TableCell></TableCell>
        <TableCell>There are no further departures at this time</TableCell>
        <TableCell></TableCell>
      </TableRow>
    </TableBody>
  );
};

const Loading = () => {
  return (
    <TableBody className="fadeIn">
      <TableRow>
        <TableCell></TableCell>
        <TableCell>
          <CircularProgress />
        </TableCell>
        <TableCell></TableCell>
      </TableRow>
    </TableBody>
  );
};

export {
  DeparturesHeader,
  DeparturesBody,
  DeparturesFooter,
  NoDepartures,
  Loading,
};
