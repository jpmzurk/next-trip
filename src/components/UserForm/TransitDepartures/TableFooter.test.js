import React from "react";
import { shallow } from 'enzyme';
import { DeparturesFooter } from "./TableModules";
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe("Icons flip when boolean state flips", () => {
  const mockClick = jest.fn();

  
  //expect footer to = snapshot

  it('should show collapse', () => {
    
    const footer = shallow(<DeparturesFooter handleExpand={mockClick} isExpanded={true} />);

    expect(footer).toMatchSnapshot()

    expect(footer.find('[name="collapse"]').exists()).toBeTruthy();

    const button = footer.find('[name="iconButton"]')
    button.simulate("click")
    expect(mockClick).toHaveBeenCalled();
  });

  it('should show expand', () => {
    const footer = shallow(<DeparturesFooter handleExpand={jest.fn()} isExpanded={false} />);

    expect(footer.find('[name="expand"]').exists()).toBeTruthy()
  })

});

// expect(enzymeToJson(cardItem)).toMatchSnapshot();



