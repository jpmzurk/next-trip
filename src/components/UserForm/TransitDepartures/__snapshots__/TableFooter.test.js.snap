// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons flip when boolean state flips should show collapse 1`] = `ShallowWrapper {}`;
